src/highdicom/sr/templates.py

```diff
@@ -1955,19 +3248,454 @@ def __init__(
+            be filtered
+
+        Returns
+        -------
+        List[highdicom.sr.SubjectContext]
+           Subject contexts
+
+        """  # noqa: E501
+        root_item = self[0]
+        matches = [
+            (i + 1, item) for i, item in enumerate(root_item.ContentSequence)
+            if item.name == codes.DCM.SubjectClass
+        ]
+        subject_contexts = []
+        for i, (index, item) in enumerate(matches):
+            if subject_class is not None:
+                if item.value != subject_class:
+                    continue
+            try:
+                next_index = matches[i + 1][0]
+            except IndexError:
+                next_index = -1
+            if item.value == codes.DCM.Specimen:
+                attributes = SubjectContextSpecimen.from_sequence(
+                    sequence=root_item.ContentSequence[index:next_index]
+                )
+            elif item.value == codes.DCM.Fetus:
+                attributes = SubjectContextFetus.from_sequence(
+                    sequence=root_item.ContentSequence[index:next_index]
+                )
+            elif item.value == codes.DCM.Device:
+                attributes = SubjectContextDevice(
+                    sequence=root_item.ContentSequence[index:next_index]
+                )
+            else:
+                raise ValueError('Unexpected subject class "{item.meaning}".')
+            context = SubjectContext(
+                subject_class=item.value,
+                subject_class_specific_context=attributes
+            )
+            subject_contexts.append(context)
+        return subject_contexts
+
+    def get_planar_roi_measurement_groups(
+        self,
+        tracking_uid: Optional[str] = None,
+        finding_type: Optional[Union[CodedConcept, Code]] = None,
+        finding_site: Optional[Union[CodedConcept, Code]] = None,
+        reference_type: Optional[Union[CodedConcept, Code]] = None,
+        graphic_type: Optional[
+            Union[GraphicTypeValues, GraphicTypeValues3D]
+        ] = None,
+        referenced_sop_instance_uid: Optional[str] = None,
+        referenced_sop_class_uid: Optional[str] = None
+    ) -> List[PlanarROIMeasurementsAndQualitativeEvaluations]:
+        """Get imaging measurement groups of planar regions of interest.
+
+        Finds (and optionally filters) content items contained in the
+        CONTAINER content item "Measurement group" as specified by TID 1410
+        "Planar ROI Measurements and Qualitative Evaluations".
+
+        Parameters
+        ----------
+        tracking_uid: str, optional
+            Unique tracking identifier
+        finding_type: Union[highdicom.sr.CodedConcept, pydicom.sr.coding.Code], optional
+            Finding
+        finding_site: Union[highdicom.sr.CodedConcept, pydicom.sr.coding.Code], optional
+            Finding site
+        reference_type: Union[highdicom.sr.CodedConcept, pydicom.sr.coding.Code], optional
+            Type of referenced ROI
```

Comment:
I think this needs more explanation for people to be able to understand what it really means. Probably an explicit list of possible values in the docstring is necessary, and perhaps a check on the input to make sure it makes sense (to give users feedback if they are passing something innappropriate)